_content/tour/ukr/composition-pollution.article

```diff
@@ -1,86 +1,86 @@
-reasonable to do so. There is a cost of using interfaces: a level of indirection
-and allocation when we store concrete values inside of them. Unless the cost of
-the allocation is worth what I’m gaining by decoupling, you shouldn't be using interfaces.
+Ні. Ми не маємо використовувати інтерфейси. Ми використовуємо інтерфейси тоді, коли в цьому є практичний зміст,
+або коли це доцільно. У використанні інтерфейсів є своя ціна: певний рівень опосередкованості
+та додаткове виділення памʼяті, коли ми зберігаємо значення всередині. Якщо ціна додаткового виділення памʼяті
+не переважає того, що ми отримуємо цим опосередкуванням, то використання інтерфейсів треба уникати.
 
-"I need to be able to test my code so I need to use interfaces".
+"Я маю тестувати свій код, тому мені треба використовувати інтерфейси".
 
-No. you must design my API for the user first, not my test. If the API is not testable,
-You should question if it’s usable. There are different layers of API’s as well. The
-lower level unexported API’s can and should focus on testability. The higher level
-exported API’s need to focus on usability.
+Ні. Ви маєте створювати свій API в першу чергу для користувача, а не для тестів. Якщо API неможливо відтестувати,
+вам треба задатися питанням, чи їм можна користуватися. Також є різні рівні API.
+Низькорівневі, не експортовані назовні API, можуть і мають фокусуватися на зручності тестування.
+Високорівневі, експортовані назовні API, мають фокусуватися на зручності використання.
 
-Functions that accept raw data in and return raw data out are the most testable.
-Separate the data transformation from where the data comes from and where it is
-going. This is a refactoring exercise you need to perform during the engineering
-coding cycle.
+Функції, які приймають сирі дані всередину і повертають сирі дані назовні легше за все тестувати.
+Відокремлюйте процес трансформації даних від того звідки дані приходять і куди
+відправляються. Це та вправа з рефакторингу, яку вам треба робити під час інженерного
+циклу розробки.
 
-Below is an example that creates interface pollution by improperly using an
-interface when one is not needed.
+Нижче можна побачити приклад, який створює забруднення інтерфейсами, використовуючи неналежним чином інтерфейс,
+коли він не потрібен.
 
     type Server interface {
         Start() error
         Stop() error
         Wait() error
     }
 
-The Server interface defines a contract for TCP servers. The problem here is I
-don’t need a contract, you need an implementation. There will only be one implementation
-as well, especially since you are the one implementing it. You do not need someone else
-to implement this for you.
+Інтерфейс Server визначає контракт для TCP серверу. Проблема тут полягає у тому, що
+вам не потрібен контракт, вам потрібна імплементація. Буде існувати лише одна імплементація,
+особливо якщо ви саме той/та хто буде її імплементувати. Вам не потрібно, щоб хтось інший
```

Comment:
```suggestion
особливо якщо ви саме той/та, хто буде її імплементувати. Вам не потрібно, щоб хтось інший
```

https://onlinecorrector.com.ua/uk/%D0%BA%D0%BE%D0%BC%D0%B0-%D0%BC%D1%96%D0%B6-%D1%87%D0%B0%D1%81%D1%82%D0%B8%D0%BD%D0%B0%D0%BC%D0%B8-%D1%81%D0%BA%D0%BB%D0%B0%D0%B4%D0%BD%D0%BE%D0%B3%D0%BE-%D1%80%D0%B5%D1%87%D0%B5%D0%BD%D0%BD%D1%8F